File: ddg.js
const https = require('https');
const fs = require('fs');

function search(query) {
  return new Promise(res => {
    https.get('https://html.duckduckgo.com/html/?q=' + encodeURIComponent(query + ' poster'), {headers: {'User-Agent': 'Mozilla/5.0'}}, resp => {
      let d=''; resp.on('data', c=>d+=c);
      resp.on('end', () => {
        const m = d.match(/<img class="result__icon__img" src="\/\/external-content\.duckduckgo\.com\/iu\/\?u=([^&"]+)/);
        if(m) res(decodeURIComponent(m[1])); else res(null);
      });
    });
  });
}

const download = (url, dest) => {
  return new Promise((resolve, reject) => {
    https.get(url, { headers: { 'User-Agent': 'Mozilla/5.0' } }, (res) => {
      if(res.statusCode===301||res.statusCode===302) return download(res.headers.location, dest).then(resolve).catch(reject);
      if(res.statusCode !== 200) return reject('Status ' + res.statusCode);
      const file = fs.createWriteStream(dest);
      res.pipe(file);
      file.on('finish', () => resolve(dest));
    }).on('error', reject);
  });
};

async function run() {
  const qs = {
    'kalki': 'Kalki 2898 AD official poster vertical',
    'pushpa': 'Pushpa 2 The Rule official poster vertical',
    'dune': 'Dune Part Two IMAX poster',
    'ind-aus': 'India vs Australia cricket action shot vertical',
    'csk-mi': 'CSK MS Dhoni IPL vertical',
    'mi-dc': 'Mumbai Indians Rohit Sharma IPL vertical',
    'pkl': 'Pro Kabaddi League action vertical',
    'edsheeran': 'Ed Sheeran concert tour poster vertical'
  };
  
  for(let [name, q] of Object.entries(qs)) {
    const url = await search(q);
    console.log(name + ': ' + url);
    if(url) {
      try {
        await download(url, 'frontend/public/highres/' + name + '.jpg');
        console.log('Downloaded ' + name);
      } catch(e) { console.log('Failed ' + name + ': ' + e); }
    }
  }
}
run();

File: ddg.py
import urllib.request
import re
import ssl
import sys

ctx = ssl.create_default_context()
ctx.check_hostname = False
ctx.verify_mode = ssl.CERT_NONE

def search(query):
    req = urllib.request.Request(
        'https://html.duckduckgo.com/html/?q=' + urllib.parse.quote(query + ' poster'),
        headers={'User-Agent': 'Mozilla/5.0'}
    )
    try:
        html = urllib.request.urlopen(req, context=ctx).read().decode('utf-8')
        m = re.search(r'<img class="result__icon__img" src="//external-content\.duckduckgo\.com/iu/\?u=([^&"]+)', html)
        if m:
            return urllib.parse.unquote(m.group(1))
    except Exception as e:
        pass
    return None

def download(url, dest):
    req = urllib.request.Request(url, headers={'User-Agent': 'Mozilla/5.0'})
    try:
        data = urllib.request.urlopen(req, context=ctx).read()
        with open(dest, 'wb') as f:
            f.write(data)
        print("Downloaded " + dest)
    except Exception as e:
        print("Failed " + dest + ": " + str(e))

qs = {
    'kalki': 'Kalki 2898 AD official poster vertical',
    'pushpa': 'Pushpa 2 The Rule official poster vertical',
    'dune': 'Dune Part Two IMAX poster',
    'ind-aus': 'India vs Australia cricket action shot vertical',
    'csk-mi': 'CSK MS Dhoni IPL vertical',
    'mi-dc': 'Mumbai Indians Rohit Sharma IPL vertical',
    'pkl': 'Pro Kabaddi League action vertical',
    'rcb-kkr': 'Virat Kohli RCB action vertical'
}

for name, q in qs.items():
    url = search(q)
    print(name + ": " + str(url))
    if url:
        download(url, "frontend/public/" + name + "_hr.jpg")
